torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2109,18 +2109,24 @@ def sample_inputs_singular_matrix_factors(op_info, device, dtype, requires_grad=
     make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
     batches = [(), (0, ), (2, ), (1, 1)]
     size = [1, 5, 10]
-
     for batch, m, n in product(batches, size, size):
-        for k in range(min(3, m, n)):
-            a = make_arg((*batch, m, k))
-            b = make_arg((*batch, n, k))
-            yield SampleInput(a, b, **kwargs)
+        k = 2
+        a = make_arg((*batch, m, k))
+        b = make_arg((*batch, n, k))
+        yield a, b
+    yield make_arg((3, 1)), make_arg((2, 1))
+    yield make_arg((1, 3, 2)), make_arg((1, 4, 2))
```

Comment:
This change set belongs to https://github.com/pytorch/pytorch/pull/127199 . Here we should have
```suggestion
        for k in range(min(3, m, n)):
            a = make_arg((*batch, m, k))
            b = make_arg((*batch, n, k))
            yield a, b
```